manual/src/html_processing/js/scroll.js

```diff
@@ -79,7 +79,7 @@ function setSmooth () {
 		if (! target) {
 		    console.log ("Error, no target for id=" + id);
 		    target = container; }
-		let top = container.scrollTop;
+		let top = Math.max(window.pageYOffset, document.documentElement.scrollTop, document.body.scrollTop);
```

Comment:
Do we need to keep support for IE < 9 ?  Using just `window.pageYOffset` seems simpler to me.

Reply:
I don't have any specific knowledge on this point, I just noticed that the current version does not work correctly on my firefox. If you think `window.pageYOffset` is enough we can go for it, on the other hand I don't really see why it hurts to keep the longer `Math.max(window.pageYOffset, document.documentElement.scrollTop, document.body.scrollTop)` 

(I also saw people writing `window.pageYOffset || document.documentElement.scrollTop ||document.body.scrollTop` but as an ocamler this disturbs me;) )

Reply:
The current version seems to work for me on Firefox, Firefox Android and Chrome? The scrolling from `https//ocaml.org/manual/objectexamples.html#s%3Arecursive-classes` to the next section (binary methods) works fine for example.

Reply:
if I go for instance to https://v2.ocaml.org/api/Unix.html#1_Basicfileinputoutput
scroll a little bit with the mouse, and click again on [Basic file input/output] in the left side-bar menu, it will jump to the section instead of the expected smooth scrolling.

Compare with https://v2.ocaml.org/api/Unix.html#1_Errorreport,
if you move away a tiny bit and click again, smooth scrolling is used as expected.

Reply:
> https//ocaml.org/manual/objectexamples.html#s%3Arecursive-classes

true, this page works for me too...

Reply:
Yes, indeed the issue is restricted to ocamldoc pages:  the problem is that the `DOCTYPE` generated by ocamldoc is non-standard and makes browsers switch to their `quirk` mode. Fixing the `DOCTYPE` removes the issue.

Reply:
great find. so if we don't care about backward compatibility, I guess we can disregard this PR and simply fix the DOCTYPE

Reply:
Since the smooth scrolling feature is an optional quality of life improvement, I think that there is indeed more value in keeping the code simple than making it backward compatible with old browser versions.